chapters/operatorsandexpressions.tex

```diff
@@ -20,15 +21,30 @@ \section{Expressions}\label{expressions}
 The built-in array functions are given in \cref{array-dimension-lower-and-upper-index-bounds} and other built-in operators in \cref{built-in-intrinsic-operators-with-function-syntax}.
 
+
 \section{Operator Precedence and Associativity}\label{operator-precedence-and-associativity}
 
-Operator precedence determines the order of evaluation of operators in an expression.
-An operator with higher precedence is evaluated before an operator with lower precedence in the same expression.
+Operator precedence determines the implicit subexpression structure of expressions with operators.
+(Explicit subexpression structure can be expressed by wrapping the subexpression in parentheses.)
+An operator with higher precedence ties harder to its operands than an operator with lower precedence.
+For example, `\lstinline!*!' having higher precedence than `\lstinline!+!' means that \lstinline!1 + 2 * 3! is implicitly structured as \lstinline!1 + (2 * 3)!.
+
+Precedence group associativity is used to determine the implicit subexpression structure when operators belong to the same group of equal precedence.
+Left associativity means that subexpressions are formed from left to right.
+For example, left associativity of binary additive operators means that \lstinline!1 - 2 - 3! is implicitly structured as \lstinline!(1 - 2) - 3!.
+A precedence group may also be non-associative, meaning that there is no implicit subexpression structure defined based on associativity.
+For example, non-associativity of relational operators means that \lstinline!1 < 2 < 3! is an invalid expression.
+Note that the operators don't need to be identical for associativity to matter; also \lstinline!1 == 2 < 3! is invalid, and \lstinline!1 - 2 + 3! is implicitly structured as \lstinline!(1 - 2) + 3!.
+Also note that the non-associative array range in Modelica can be used with either two or three operands separated by `\lstinline!:!', meaning that \lstinline!1 : 2 : 5! is one valid ternary use of the operator rather than two invalid binary uses of the operator.
 
-The following table presents all the expression operators in order of precedence.
+Note that at the parsing stage -- which is where the here defined operator precedence and associativity matters -- the subexpression structure is fixed.
+Since Modelica tools have the freedom to symbolically manipulate expressions, this subexpression structure cannot be expected to reflect order of evaluation, compare \cref{evaluation-order}.
+
+The following table presents the precedence and associativity of all the expression operators, consistent with and complementing information that can be derived from the Modelica grammar in \cref{modelica-concrete-syntax}.
 \begin{table}[H]
 \caption{%
-Operators in order of precedence from highest to lowest, as derived from the Modelica grammar in \cref{modelica-concrete-syntax}.
+Operators in order of precedence from highest to lowest.
+Operators with the same precedence occur at the same table row.
```

Comment:
What I find unclear in both the existing table and in this PR is what is considered a row of the table. Do horizontal lines mean anything or are they just to make it look better? It almost suggests that only what is separated by lines are considered to be rows. For example, are additive and elementwise additive are in the same row or not? I thought they would be (both can be used on a scalar so it's not illegal to combine them). But they are in the different rows, if we look only  at the actual rows.

Reply:
I agree that this is confusing, and as I understand it only "rows" separated by horizontal lines should be seen as different rows.

Reply:
I have applied @HansOlsson's suggestion.  @eshmoylova, please mark as resolved if this looks good to you.